bin/worker.js

```diff
@@ -0,0 +1,139 @@
+#!/usr/bin/env node
+var debug             = require('debug')('github:worker');
+var base              = require('taskcluster-base');
+var Promise           = require('promise');
+var exchanges         = require('../lib/exchanges');
+var worker            = require('../lib/worker');
+var _                 = require('lodash');
+var taskcluster       = require('taskcluster-client');
+var GitHubAPI         = require('github');
+
+/** Launch worker */
+var launch = async function(profile) {
+  debug("Launching with profile: %s", profile);
+
+  // Load configuration
+  var cfg = base.config({
+    defaults:     require('../config/defaults'),
+    profile:      require('../config/' + profile),
+    envs: [
+      'pulse_username',
+      'pulse_password',
+      'taskclusterGithub_publishMetaData',
+      'taskcluster_credentials_clientId',
+      'taskcluster_credentials_accessToken',
+      'aws_accessKeyId',
+      'aws_secretAccessKey',
+      'influx_connectionString',
+      'webhook_secret'
+    ],
+    filename:     'taskcluster-github'
+  });
+
+  // Create a single connection to the GithubAPI to pass around
+  var githubAPI = new GitHubAPI(cfg.get('github:config'));
+  githubAPI.authenticate(cfg.get('github:credentials'));
+
+  // A context to be passed into message handlers
+  let context = {cfg, githubAPI};
+
+  // Create a default stats drain, which just prints to stdout
+  let statsDrain = {
+      addPoint: (...args) => {debug("stats:", args)}
+  }
+
+  // Create InfluxDB connection for submitting statistics
+  let influxConnectionString = cfg.get('influx:connectionString')
+  if (influxConnectionString) {
+      statsDrain = new base.stats.Influx({
+        connectionString:   influxConnectionString,
+        maxDelay:           cfg.get('influx:maxDelay'),
+        maxPendingPoints:   cfg.get('influx:maxPendingPoints')
+      });
+  } else {
+      debug("Missing influx_connectionString: stats collection disabled.")
+  }
+
+  // Start monitoring the process
+  base.stats.startProcessUsageReporting({
+    drain:      statsDrain,
+    component:  cfg.get('taskclusterGithub:statsComponent'),
+    process:    'worker'
+  });
+
+  let pulseCredentials = cfg.get('pulse')
+  if (pulseCredentials.username && pulseCredentials.password) {
```

Comment:
Two things I've seen in some of the taskcluster related things is either using "assert" to verify something is in the config if it would result in throwing an exception:
```js
assert(pulseCredentails.username, 'Username must be supplied for Pulse connection')
```
or 
instead of wrapping the creation of the pulselistener, throw early...
```js
if (!pulseCredentials.username || !pulseCredentials.password) throw new Error('Missing pulse credentails');
```

Either way, nothing critical